source/extensions/filters/network/dubbo_proxy/protocol.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/common/fmt.h"
+#include "common/singleton/const_singleton.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+/**
+ * Names of available Protocol implementations.
+ */
+class ProtocolNameValues {
+public:
+  // Dubbo protocol
+  const std::string Dubbo = "dubbo";
+};
+
+typedef ConstSingleton<ProtocolNameValues> ProtocolNames;
+
+// Supported serialization type
+enum class SerializationType : uint8_t {
+  Hessian = 2,
+  Json = 6,
+};
+
+// Message Type
+enum class MessageType : uint8_t {
+  Response = 0,
+  Request = 1,
+};
+
+/**
+ * Dubbo protocol response status types.
+ * See org.apache.dubbo.remoting.exchange.codec
+ */
+enum class ResponseStatus : uint8_t {
+  Ok = 20,
+  ClientTimeout = 30,
+  ServerTimeout = 31,
+  BadRequest = 40,
+  BadResponse = 50,
+  ServiceNotFound = 60,
+  ServiceError = 70,
+  ServerError = 80,
+  ClientError = 90,
+  ServerThreadpoolExhaustedError = 100,
+};
+
+class Message {
+public:
+  virtual ~Message() {}
+  virtual MessageType messageType() const PURE;
+  virtual int32_t bodySize() const PURE;
+  virtual bool isEvent() const PURE;
+  virtual int64_t requestId() const PURE;
+  virtual std::string toString() const PURE;
+};
+
+class RequestMessage : public virtual Message {
+public:
+  virtual ~RequestMessage() {}
+  virtual SerializationType serializationType() const PURE;
+  virtual bool isTwoWay() const PURE;
+};
+
+typedef std::unique_ptr<RequestMessage> RequestMessagePtr;
+
+class ResponseMessage : public virtual Message {
+public:
+  virtual ~ResponseMessage() {}
+  virtual ResponseStatus responseStatus() const PURE;
+};
+
+typedef std::unique_ptr<ResponseMessage> ResponseMessagePtr;
+
+/**
+ * ProtocolCallbacks are Dubbo protocol-level callbacks.
+ */
+class ProtocolCallbacks {
+public:
+  virtual ~ProtocolCallbacks() {}
+  virtual void onRequestMessage(RequestMessagePtr&& req) PURE;
+  virtual void onResponseMessage(ResponseMessagePtr&& res) PURE;
+};
+
+/**
+ * See https://dubbo.incubator.apache.org/#!/docs/dev/implementation.md?lang=en-us
```

Comment:
This link doesn't work (redirected to https://dubbo.incubator.apache.org/en-us/), do you have a valid one?